kafkaesk/ext/logging/handler.py

```diff
@@ -0,0 +1,111 @@
+from .formatter import PydanticLogModel
+from .record import PydanticLogRecord
+from pydantic import BaseModel
+from typing import IO
+from typing import Optional
+from typing import Tuple
+
+import asyncio
+import kafkaesk
+import logging
+import sys
+
+
+class InvalidLogFormat(Exception):
+    ...
+
+
+class PydanticStreamHandler(logging.StreamHandler):
+    def __init__(self, stream: Optional[IO[str]] = None):
+        super().__init__(stream=stream)
+
+    def format(self, record: logging.LogRecord) -> str:
+        message = super().format(record)
+
+        if isinstance(message, BaseModel):
+            message = message.json()
+
+        return message
+
+
+class LogQueue:
+    def __init__(self, app: kafkaesk.Application):
+        self._queue: asyncio.Queue[Tuple[str, BaseModel]] = asyncio.Queue()
+        self._app = app
+
+    async def run(self) -> None:
+        while True:
+            stream, message = await self._queue.get()
+            try:
+                await self._app.publish(stream, message)
+            except Exception as err:
+                print(err)
+            finally:
+                self._queue.task_done()
```

Comment:
is this really needed since no one - unless I'm missing something - is synchronizing through a `join`[1] ?


[1] https://docs.python.org/3/library/asyncio-queue.html#asyncio.Queue.join

Reply:
Nope, not needed.